flang/lib/Lower/OpenMP/OpenMP.cpp

```diff
@@ -2017,36 +2166,47 @@ static void genOMP(Fortran::lower::AbstractConverter &converter,
       semaCtx);
   mlir::Location currentLocation = converter.genLocation(directive.source);
 
+  ConstructQueue queue{
+      buildConstructQueue(converter.getFirOpBuilder().getModule(), semaCtx,
+                          eval, directive.source, directive.v, clauses)};
+
   switch (directive.v) {
```

Comment:
Is there a reason why this can't be moved into `genOMPDispatch` as well? We won't find compound constructs here, but being consistent should simplify the reasoning about how lowering is done.

Reply:
I left it out because I expected a comment from you about why use the dispatch if these are simple constructs...  lol

Reply:
That's funny, you mispredicted a branch there! I think it's fine to make a follow-up PR for that.